src/main/java/com/github/mizool/jediscache/JedisCache.java

```diff
@@ -98,19 +99,25 @@ public V get(K key)
     @Override
     public Map<K, V> getAll(Set<? extends K> keys)
     {
-        Map<byte[], byte[]> allEntries;
+        List<byte[]> allEntries;
         try (Jedis jedis = obtainJedis())
         {
-            allEntries = jedis.hgetAll(jedisCacheName);
+            allEntries = jedis.hmget(jedisCacheName, keys.stream().map(converter::serialize).toArray(byte[][]::new));
         }
         ImmutableMap.Builder<K, V> builder = ImmutableMap.builder();
-        keys.forEach(key -> {
-            byte[] serializedKey = converter.serialize(key);
-            if (allEntries.containsKey(serializedKey))
+        K key = null;
+        for (byte[] entry : allEntries)
```

Comment:
I had a hard time to understand the "every even element in the list is a key and every odd element is a value" mechanic, that's hidden behind `key` being `null` or not.

I wonder if it becomes more clear if we use either of the following approaches...

* an int stream that ranges from zero to `allEntries.size() - 1` that skips every odd element and addresses keys and values in pairs:
```java
Map<K, V> result = IntStream.range(0, allEntries.size())
    .filter(i -> i % 2 == 0)
    .collect(ImmutableMap.toImmutableMap(i -> {
        byte[] key = allEntries.get(i);
        K convertedKey = converter.deserialize(key, keyClass);
        return convertedKey;
    }, i -> {
        byte[] value = allEntries.get(i + 1);
        V convertedValue = converter.deserialize(value, valueClass);
        return convertedValue;
    }));
```
_In this example, the key and value function should probably be extracted to methods._

* an old fashioned for loop with an integer variable that is incremented in steps of two that is used to address keys and values in pairs:
```java
ImmutableMap.Builder<K, V> builder = ImmutableMap.builder();
for (int i = 0; i < allEntries.size(); i = i + 2)
{
    byte[] key = allEntries.get(i);
    K convertedKey = converter.deserialize(key, keyClass);
    
    byte[] value = allEntries.get(i + 1);
    V convertedValue = converter.deserialize(value, valueClass);
    
    builder.put(convertedKey, convertedValue);
}
Map<K, V> result = builder.build();
```

I think both approaches are a little more robust than the current one, although I can't decide between them. The first one is a bit odd with it's filter that does a modulo for no obvious reason. The second one looks a lot like a normal for-i loop and the detail that we increment by two is not _that_ obvious.

Reply:
...or we use the third method, we turn the `Stream<byte[]>` into a `Stream<Map.Entry<byte[], byte[]>>` using `.flatMap()`:
```java
new Function<E, Stream<Map.Entry<E, E>>>()
{
    private E key;

    @Override
    public Stream<Map.Entry<E, E>> apply(@NonNull E e)
    {
        if (key == null)
        {
            key = e;
            return Stream.of();
        }
        else
        {
            key = null;
            return Stream.of(Maps.<E, E>immutableEntry(key, e));
        }
    }
}
```